kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetFilterUtils.java

```diff
@@ -334,6 +341,61 @@ private static Optional<FilterPredicate> convertAndToParquetFilter(
         return rightFilter;
     }
 
+    private static Optional<FilterPredicate> convertIsNullIsNotNull(
+            Map<Column, ParquetField> parquetFieldMap,
+            Predicate deltaPredicate,
+            boolean isNotNull) {
+        Expression child = getUnaryChild(deltaPredicate);
+        if (!(child instanceof Column)) {
+            return visitUnsupported(deltaPredicate, "IS NULL predicate must have a column input.");
+        }
+
+        Column column = (Column) child;
+        ParquetField parquetField = parquetFieldMap.get(column);
+        if (parquetField == null) {
+            return visitUnsupported(
+                    deltaPredicate,
+                    "Column used in predicate does not exist in the parquet file.");
+        }
+
+        String columnPath = ColumnPath.get(column.getNames()).toDotString();
+        // Parquet filter keeps records if their value is equal to the provided value.
+        // Nulls are treated the same way the java programming language does.
+        // For example: eq(column, null) will keep all records whose value is null. eq(column, 7)
+        // will keep all records whose value is 7, and will drop records whose value is null
+        // NOTE: this is different from how some query languages handle null.
+        switch (parquetField.primitiveType.getPrimitiveTypeName()) {
+            case BOOLEAN:
+                BooleanColumn booleanColumn = booleanColumn(columnPath);
+                return Optional.of(isNotNull ?
```

Comment:
is there a parent class of BooleanColumn and IntColumn?

Could we create a helper method:

```java
Optional<FilterPredicate> createIsNullOrIsNotNullPredicate(Column column, boolean isNotNull) {
  return Optional.of(isNotNull ? FilterApi.notEq(column, null) : FilterApi.eq(column, null));
}
```